bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,819 @@
+
+                JsonElement jsonTree = parser.parse(reader);
+
+                logger.debug("Communication result [{}]", response.getStatus());
+
+                if (response.getStatus() != 200) {
+                    logger.error("Communication error by getDatapoint [{}]", response.getStatus());
+                }
+
+                // check the output
+                if (jsonTree != null) {
+                    if (jsonTree.isJsonObject()) {
+                        JsonObject jsonObject = jsonTree.getAsJsonObject();
+
+                        jsonObject = jsonObject.getAsJsonObject(sysApUID);
+                        JsonArray jsonValueArray = jsonObject.getAsJsonArray("values");
+
+                        JsonElement element = jsonValueArray.get(0);
+                        String value = element.getAsString();
+
+                        if (value.isEmpty()) {
+                            value = "0";
+                        }
+
+                        return value;
+                    }
+                }
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            logger.error("Communication error by getDatapoint [{}]", e.getMessage());
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+
+            return new String("0");
+        }
+
+        return new String("0");
+    }
+
+    /**
+     * Method to set datapoint values in channels
+     *
+     * @author Andras Uhrin
+     *
+     */
+    public boolean setDatapoint(@Nullable String deviceId, @Nullable String channel, @Nullable String datapoint,
+            String valueString) {
+        if ((deviceId == null) || (channel == null) || (datapoint == null)) {
+            return false;
+        }
+
+        String url = baseUrl + "/rest/datapoint/" + sysApUID + "/" + deviceId + "." + channel + "." + datapoint;
+        try {
+            Request req = httpClient.newRequest(url);
+
+            req.content(new StringContentProvider(valueString));
+            req.method(HttpMethod.PUT);
+
+            logger.debug("Set datapoint by url: {} value: {}", url, valueString);
+
+            ContentResponse response = req.send();
+
+            logger.debug("Communication result [{}]", response.getStatus());
+
+            if (response.getStatus() != 200) {
+                logger.error("Communication error by setDatapoint [{}]", response.getStatus());
+
+                restartHttpConnection();
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            logger.error("Communication error by setDatapoint [{}]", e.getMessage());
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+
+        return true;
+    }
+
+    /**
+     * Method to process socket events
+     *
+     * @author Andras Uhrin
+     *
+     */
+    @SuppressWarnings("deprecation")
```

Comment:
Why?

Reply:
@SuppressWarnings("deprecation") is needed due to jsonParser.